core/lib/mno_enterprise/concerns/controllers/auth/confirmations_controller.rb

```diff
@@ -92,6 +92,9 @@ def finalize
     end
 
     if resource.errors.empty?
+      if params[:tos]
```

Comment:
The form can be submitted only when the TOS are accepted so it souldn't be required. 
However, checking the exact string will improve clarity at no cost, I'm adding it.